paddle/fluid/operators/mkldnn/concat_mkldnn_op.cc

```diff
@@ -156,12 +167,9 @@ class ConcatMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
         paddle::framework::ToMKLDNNDataType(multi_input[0]->type());
 
     ConcatPrimitiveFactory<T> prim_creator;
-    // If one of the multiple inputs of concat has an input size of 0, the
-    // actual size of the multi_input will change
     std::string key = platform::CreateKey(
-        dev_ctx, paddle::framework::vectorize<int>(multi_input[0]->dims()),
-        multi_input.size(), ctx.OutputName("Out"), dt,
-        platform::ThreadIDasStr());
+        dev_ctx, GetDimsForKey(multi_input), multi_input.size(),
+        ctx.OutputName("Out"), dt, platform::ThreadIDasStr());
```

Comment:
If possible please remove platform::ThreadIDasStr() , it is added  in ExtendKeyWithThreadInfoIfNeeded().

Reply:
Done